WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingViewModel.swift

```diff
@@ -47,19 +47,36 @@ class StoreOnboardingViewModel: ObservableObject {
     private let placeholderTasks: [StoreOnboardingTaskViewModel] = Array(repeating: StoreOnboardingTaskViewModel.placeHolder(),
                                                                          count: 3)
 
-    /// - Parameter isExpanded: Whether the onboarding view is in the expanded state. The expanded state is shown when the view is in fullscreen.
+    private let defaults: UserDefaults
+
+    private let onAllOnboardingTasksCompleted: (() -> Void)?
+
+    /// Initializer
+    /// - Parameters:
+    ///   - isExpanded: Whether the onboarding view is in the expanded state. The expanded state is shown when the view is in fullscreen.
+    ///   - siteID: siteID
+    ///   - stores: StoresManager
+    ///   - userDefaults: UserDefaults for storing when all onboarding tasks are completed
+    ///   - onAllOnboardingTasksCompleted:Callback to fire when all onboarding tasks are completed
     init(isExpanded: Bool,
          siteID: Int64,
-         stores: StoresManager = ServiceLocator.stores) {
+         stores: StoresManager = ServiceLocator.stores,
+         defaults: UserDefaults = .standard,
+         onAllOnboardingTasksCompleted: (() -> Void)? = nil) {
         self.isExpanded = isExpanded
         self.siteID = siteID
         self.stores = stores
         self.state = .loading
+        self.defaults = defaults
+        self.onAllOnboardingTasksCompleted = onAllOnboardingTasksCompleted
     }
 
     func reloadTasks() async {
         await update(state: .loading)
         let tasks = try? await loadTasks()
+        if let tasks {
```

Comment:
I am afraid we cannot do this because,
- I need the `tasks` value to be available for [this line](https://github.com/woocommerce/woocommerce-ios/pull/9129/files/920d09a310aa0d29872f99bcb5365c34781649b6#diff-71c880a9f74f980fa43772be01c43133abf24d84c81a79946b54d53db72209e2R80).
- And that line cannot be moved inside the `if` condition because I want to use `taskViewModels` if `tasks` is nil.

Reply:
> * And that line cannot be moved inside the `if` condition because I want to use `taskViewModels` if `tasks` is nil.

Can this be called in the `else` block then? just a non-blocking minor suggestion.

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/9129/commits/bac66bfc50c0ecda70e9f1cf1c5335d6a048d8b3